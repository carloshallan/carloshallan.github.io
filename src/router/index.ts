import Vue from 'vue'
import VueRouter, { RouteConfig } from 'vue-router'

Vue.use(VueRouter)

const HomeComponent = () => import('@/views/Home.vue')
const WorkComponent = () => import('@/views/Work.vue')
const PostComponent = () => import('@/views/PostView.vue')

const routes: Array<RouteConfig> = [
  {
    path: '/',
    name: 'Home',
    component: HomeComponent
  },
  {
    path: '/work',
    name: 'Work',
    component: WorkComponent
  },
  {
    path: '/posts/:slug',
    name: 'post',
    component: PostComponent
  }
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes
})

export default router
