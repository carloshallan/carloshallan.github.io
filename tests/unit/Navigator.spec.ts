import { mount } from '@vue/test-utils'
import Nav from '@/components/Navigator.vue'
import router from '@/router'

describe('Navigator.vue', () => {
  it('Render Navigator', () => {
    const navigator = mount(Nav, { router })

    expect(navigator.element).toMatchSnapshot()
  })
})
