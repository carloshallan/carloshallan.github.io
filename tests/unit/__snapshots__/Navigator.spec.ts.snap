// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator.vue Render Navigator 1`] = `
<nav>
  <li
    class="router-link-exact-active active"
  >
    
    Home
  
  </li>
  <li
    class=""
  >
    
    Work
  
  </li>
  <li
    class=""
  >
    
    post
  
  </li>
</nav>
`;
